### ISSUE TITLE
docker build for ubuntu 15.04 does not work

### ISSUE THREAD

**Issue by despiegk** [Jumpscale/jumpscale_core7#136]
```
root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04# make
rm -rf /tmp/jumpscale/ubuntu-15.04
mkdir -p /tmp/jumpscale/ubuntu-15.04
cp -r ../../baseimage/image/* /tmp/jumpscale/ubuntu-15.04/
cp: cannot stat ‘../../baseimage/image/*’: No such file or directory
Makefile:14: recipe for target 'prepare' failed
make: *** [prepare] Error 1
root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04# ls
Dockerfile  Makefile  readme.md
root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
```

there seem to be missing files

**Comment by muhamadazmy** [comment]
You need to do

git submodule init
first to pull missing files as per the readme file

**Comment by muhamadazmy** [comment]
Also git submodule update

**Comment by despiegk** [comment]
why this extra complexity?
and I did the git submodule init?
can we make it easier please

On Fri, Sep 11, 2015 at 11:58 AM, Muhamad Azmy <notifications@github.com>
wrote:

> You need to do
>
> git submodule init
> first to pull missing files as per the readme file
> On 11 Sep 2015 08:05, "kristof de spiegeleer" <notifications@github.com>
> wrote:
>
> > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04# make
> > rm -rf /tmp/jumpscale/ubuntu-15.04
> > mkdir -p /tmp/jumpscale/ubuntu-15.04
> > cp -r ../../baseimage/image/* /tmp/jumpscale/ubuntu-15.04/
> > cp: cannot stat ‘../../baseimage/image/*’: No such file or directory
> > Makefile:14: recipe for target 'prepare' failed
> > make: *** [prepare] Error 1
> > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04# ls
> > Dockerfile Makefile readme.md
> > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
> >
> > there seem to be missing files
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/Jumpscale/jumpscale_core7/issues/136>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139504517>
> .
>



-- 
------------------------
Kristof De Spiegeleer
+971525609014
+201206927877
+32 475405474
skype: despiegk

**Comment by muhamadazmy** [comment]
Did it build after doing 'git submodule update'?
Sure, I will make it easier by include the sumbodule intiallization step in
the make file.

**Comment by despiegk** [comment]
doesnt work, I am stuck
because it was .git and if you don't have a valid key on github it will fail
i brought it to https but that seems to be wrong too
anyhow I think we also need to do git submodule update
which was missing in doc
please let me know what to do

On Fri, Sep 11, 2015 at 12:45 PM, Muhamad Azmy <notifications@github.com>
wrote:

> Did it build after doing 'git submodule update'?
> Sure, I will make it easier by include the sumbodule intiallization step in
> the make file.
> On 11 Sep 2015 12:24, "kristof de spiegeleer" <notifications@github.com>
>
> wrote:
>
> > why this extra complexity?
> > and I did the git submodule init?
> > can we make it easier please
> >
> > On Fri, Sep 11, 2015 at 11:58 AM, Muhamad Azmy <notifications@github.com
> >
> > wrote:
> >
> > > You need to do
> > >
> > > git submodule init
> > > first to pull missing files as per the readme file
> > > On 11 Sep 2015 08:05, "kristof de spiegeleer" <
> notifications@github.com>
> > > wrote:
> > >
> > > > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
> > make
> > > > rm -rf /tmp/jumpscale/ubuntu-15.04
> > > > mkdir -p /tmp/jumpscale/ubuntu-15.04
> > > > cp -r ../../baseimage/image/* /tmp/jumpscale/ubuntu-15.04/
> > > > cp: cannot stat ‘../../baseimage/image/*’: No such file or directory
> > > > Makefile:14: recipe for target 'prepare' failed
> > > > make: *** [prepare] Error 1
> > > > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
> ls
> > > > Dockerfile Makefile readme.md
> > > > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
> > > >
> > > > there seem to be missing files
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <https://github.com/Jumpscale/jumpscale_core7/issues/136>.
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139504517
> > >
> > > .
> > >
> >
> >
> >
> > --
> > ------------------------
> > Kristof De Spiegeleer
> > +971525609014
> > +201206927877
> > +32 475405474
> > skype: despiegk
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139511167
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139513983>
> .
>



-- 
------------------------
Kristof De Spiegeleer
+971525609014
+201206927877
+32 475405474
skype: despiegk

**Comment by muhamadazmy** [comment]
I guess (not sure) if you changed the submodulr to use https you need to do
git submodule deinit
git submodule update --init

I don't have my laptop atm so I can't give this a try
doesnt work, I am stuck
because it was .git and if you don't have a valid key on github it will fail
i brought it to https but that seems to be wrong too
anyhow I think we also need to do git submodule update
which was missing in doc
please let me know what to do

On Fri, Sep 11, 2015 at 12:45 PM, Muhamad Azmy <notifications@github.com>
wrote:

> Did it build after doing 'git submodule update'?
> Sure, I will make it easier by include the sumbodule intiallization step
in
> the make file.
> On 11 Sep 2015 12:24, "kristof de spiegeleer" <notifications@github.com>
>
> wrote:
>
> > why this extra complexity?
> > and I did the git submodule init?
> > can we make it easier please
> >
> > On Fri, Sep 11, 2015 at 11:58 AM, Muhamad Azmy <notifications@github.com
> >
> > wrote:
> >
> > > You need to do
> > >
> > > git submodule init
> > > first to pull missing files as per the readme file
> > > On 11 Sep 2015 08:05, "kristof de spiegeleer" <
> notifications@github.com>
> > > wrote:
> > >
> > > > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
> > make
> > > > rm -rf /tmp/jumpscale/ubuntu-15.04
> > > > mkdir -p /tmp/jumpscale/ubuntu-15.04
> > > > cp -r ../../baseimage/image/* /tmp/jumpscale/ubuntu-15.04/
> > > > cp: cannot stat ‘../../baseimage/image/*’: No such file or directory
> > > > Makefile:14: recipe for target 'prepare' failed
> > > > make: *** [prepare] Error 1
> > > > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
> ls
> > > > Dockerfile Makefile readme.md
> > > > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
> > > >
> > > > there seem to be missing files
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <https://github.com/Jumpscale/jumpscale_core7/issues/136>.
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
>
https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139504517
> > >
> > > .
> > >
> >
> >
> >
> > --
> > ------------------------
> > Kristof De Spiegeleer
> > +971525609014
> > +201206927877
> > +32 475405474
> > skype: despiegk
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
>
https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139511167
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <
https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139513983
>
> .
>



-- 
------------------------
Kristof De Spiegeleer
+971525609014
+201206927877
+32 475405474
skype: despiegk

—
Reply to this email directly or view it on GitHub
<https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139514988>
.

**Comment by despiegk** [comment]
i did that

On Fri, Sep 11, 2015 at 1:02 PM, Muhamad Azmy <notifications@github.com>
wrote:

> I guess (not sure) if you changed the submodulr to use https you need to do
> git submodule deinit
> git submodule update --init
>
> I don't have my laptop atm so I can't give this a try
>
> doesnt work, I am stuck
> because it was .git and if you don't have a valid key on github it will
> fail
> i brought it to https but that seems to be wrong too
> anyhow I think we also need to do git submodule update
> which was missing in doc
> please let me know what to do
>
> On Fri, Sep 11, 2015 at 12:45 PM, Muhamad Azmy <notifications@github.com>
> wrote:
>
> > Did it build after doing 'git submodule update'?
> > Sure, I will make it easier by include the sumbodule intiallization step
> in
> > the make file.
> > On 11 Sep 2015 12:24, "kristof de spiegeleer" <notifications@github.com>
> >
> > wrote:
> >
> > > why this extra complexity?
> > > and I did the git submodule init?
> > > can we make it easier please
> > >
> > > On Fri, Sep 11, 2015 at 11:58 AM, Muhamad Azmy <
> notifications@github.com
> > >
> > > wrote:
> > >
> > > > You need to do
> > > >
> > > > git submodule init
> > > > first to pull missing files as per the readme file
> > > > On 11 Sep 2015 08:05, "kristof de spiegeleer" <
> > notifications@github.com>
> > > > wrote:
> > > >
> > > > > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
> > > make
> > > > > rm -rf /tmp/jumpscale/ubuntu-15.04
> > > > > mkdir -p /tmp/jumpscale/ubuntu-15.04
> > > > > cp -r ../../baseimage/image/* /tmp/jumpscale/ubuntu-15.04/
> > > > > cp: cannot stat ‘../../baseimage/image/*’: No such file or
> directory
> > > > > Makefile:14: recipe for target 'prepare' failed
> > > > > make: *** [prepare] Error 1
> > > > > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
> > ls
> > > > > Dockerfile Makefile readme.md
> > > > > root@ubuntu:/opt/code/github/jumpscale/dockers/images/ubuntu15.04#
> > > > >
> > > > > there seem to be missing files
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <https://github.com/Jumpscale/jumpscale_core7/issues/136>.
> > > > >
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
>
> https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139504517
> > > >
> > > > .
> > > >
> > >
> > >
> > >
> > > --
> > > ------------------------
> > > Kristof De Spiegeleer
> > > +971525609014
> > > +201206927877
> > > +32 475405474
> > > skype: despiegk
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
>
> https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139511167
> > >
> >
> > > .
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
>
> https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139513983
> >
> > .
> >
>
>
>
> --
> ------------------------
> Kristof De Spiegeleer
> +971525609014
> +201206927877
> +32 475405474
> skype: despiegk
>
> —
> Reply to this email directly or view it on GitHub
> <
> https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139514988
> >
>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jumpscale/jumpscale_core7/issues/136#issuecomment-139516541>
> .
>



-- 
------------------------
Kristof De Spiegeleer
+971525609014
+201206927877
+32 475405474
skype: despiegk

**Comment by muhamadazmy** [comment]
I fixed the submodule issue, the submodule url shouldn't be surrounded with ' (single quotes). 

I also changed the make files so you don't have to init or update the submodule manually, the make will take care of that.

**Comment by despiegk** [comment]
cool works

On Sun, Sep 13, 2015 at 8:22 AM, Muhamad Azmy <notifications@github.com>
wrote:

> Closed #136 <https://github.com/Jumpscale/jumpscale_core7/issues/136>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jumpscale/jumpscale_core7/issues/136#event-407989017>.
>



-- 
------------------------
Kristof De Spiegeleer
+971525609014
+201206927877
+32 475405474
skype: despiegk
